json-logic/README.md

```diff
@@ -0,0 +1,62 @@
+# CovQuestions Draft
+
+This is the draft of a custom questionaire format for the CovQuestions project.
+
+## Requirements
+
+* Relatively easy to edit
+* Easy to use and implement
+* As platform-independend as possible
+* Describe questions, results, and result-computation logic
+* Questions might be shown or not depending in intermediate results
+* Support for multiple result categories, with their own results and computations
+
+## Considerations
+
+### Result Format
+
+Two possible options: 
+1. "Flat" list of results with a expressione evaluating to a boolean
+2. Result categories with an expression that evaluate to a result id with a list of results per category.
+
+We favor option 2 - with option 1 it is easy to accidentially show two results from the same category. This is impossible with 2 by design at the price of one additional layer of nesting.
+
+### Evaluation
+
+Two possible options: 
+1. Put all variables and questions into the same array and process them in order.
+2. Seperate variables and questions, process all questions in order and (re-)evaluate all variables after each update.
+
+We favor option 2 - it gives the user one less thing to worry about.
+
+#### Structuring Expressions
+
+JSON Logic is powerful, but messy to read/write. There are multiple ways to express the same logic.
+Once we have a better understanding of the task at hand, we should come up with some basic guidelines for
+formatting and structuring.
+
+### Dependencies between variables
+
+We favor a relatively simple logic and no automatic dependency resolution/re-ordering of questions or variables. 
+If things are ordered incorrectly (e.g. the skipIf in a question refers a future question), the questionaire author should discover and repair the issue during testing.
+
+### Answer Representation
+
+For each answered question, we multiple JSON-Logic variables prefixed with the question id:
+
+```
+question_id.value // string, array of string, number, boolean, date, depending on type
+question_id.count // number
```

Comment:
What does `count` mean here? Is it the length of a multiselect answer?

Reply:
Exactly. JSON Logic has no length operator on arrays, so calculating the length either requires a manual reduce. 

I thought injecting a count attribute makes things easier.

Reply:
And what's the use case behind it? Where do we use the length of the answer array?

Reply:
For example: 
"Has at least one symptom", "Has at least two symptoms".

Reply:
We might also use it to determine if an empty answer was given (count===0) vs no answer yet (count===undefined)

Reply:
I don't know if it's important right now. 
Do we maybe need another attribute to determine if a question was skipped?

Reply:
Well but if you do it this way, there is no information in the answer about things that the user did not select? 
Also, I would rename it then `selectedCount`? Because it is not clear, what count is meant (I would have just assumed its the overall length of the answers.
(That's another point, on why I dislike the multi-select)

Reply:
> Well but if you do it this way, there is no information in the answer about things that the user did not select?

That's what the `.value` would be for.

Reply:
But value only contains information about the selected items, not about the ones that were not selected

Reply:
Sorry - saw this comment too late.

Let's see if we find a use case during evaluation where we explicitly need the list of not-selected items.